# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/base/nsURLHelper.h
# Commit: b7fab4b145f7
# Full Hash: b7fab4b145f74d207f421836b32e86625f1759c5
# Author: abhishekmadan30 <148475204+abhishekmadan30@users.noreply.github.com>
# Date: 2024-03-26 21:18:53
# Regressor Bug: 1877419
# File Overlap Count: 1
# Description:
#   Bug 1877419 - Part 1: Added nested stripping of urls to copy without site tracking. r=pbz,necko-reviewers
#   
#   Depends on D204914
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D202943
# ==============================================================================

diff -r 78436da20d95 -r b7fab4b145f7 netwerk/base/nsURLHelper.h
--- a/netwerk/base/nsURLHelper.h	Tue Mar 26 14:42:07 2024 +0000
+++ b/netwerk/base/nsURLHelper.h	Tue Mar 26 14:42:07 2024 +0000
@@ -308,6 +308,8 @@
    */
   void Serialize(nsAString& aValue, bool aEncode) const;
 
+  static void SerializeString(const nsCString& aInput, nsAString& aValue);
+
   void Get(const nsAString& aName, nsString& aRetval);
 
   void GetAll(const nsAString& aName, nsTArray<nsString>& aRetval);
@@ -338,6 +340,8 @@
 
   uint32_t Length() const { return mParams.Length(); }
 
+  static void DecodeString(const nsACString& aInput, nsAString& aOutput);
+
   const nsAString& GetKeyAtIndex(uint32_t aIndex) const {
     MOZ_ASSERT(aIndex < mParams.Length());
     return mParams[aIndex].mKey;
@@ -355,7 +359,6 @@
   void Sort();
 
  private:
-  static void DecodeString(const nsACString& aInput, nsAString& aOutput);
   static void ConvertString(const nsACString& aInput, nsAString& aOutput);
   static bool ParseNextInternal(const char*& aStart, const char* aEnd,
                                 bool aShouldDecode, nsAString* aOutputName,